server/restapi/restapi.coffee

```diff
@@ -0,0 +1,59 @@
+Api = new Restivus
+	useDefaultAuth: true
+	prettyJson: true
+
+
+Api.addCollection ChatMessage
+Api.addCollection ChatRoom
+
+
+Api.addRoute 'version', authRequired: false,
+	get: ->
+		version = {api: '0.1', rocketchat: '0.5'}
+		status: 'success', versions: version
+
+Api.addRoute 'publicRooms', authRequired: true,
+	get: ->
+		rooms = ChatRoom.find({ t: 'c' }, { sort: { msgs:-1 } }).fetch()
+		if rooms
+			status: 'success', rooms: rooms
+		else
+			statusCode: 404
+			body: status: 'fail', message: 'Rooms not found'
+
+# join a room
+Api.addRoute 'rooms/:id/join', authRequired: true,
+	post: ->
+		Meteor.runAsUser this.userId, () =>
+			Meteor.call('joinRoom', @urlParams.id)
+		status: 'success'   # need to handle error
+
+
+# leave a room
+Api.addRoute 'rooms/:id/leave', authRequired: true,
+	post: ->
+		Meteor.runAsUser this.userId, () =>
+			Meteor.call('leaveRoom', @urlParams.id)
+		status: 'success'   # need to handle error
+
+
+# get messages in a room
+Api.addRoute 'rooms/:id/messages', authRequired: true,
+	get: ->
```

Comment:
Is a call to 'canAccessRoom' going to be strong enough? 
API is on the server.   But on the client, a user seems to be able to 'visit' any room and see the messages.
I am weary about emulating client behaviour at this stage on API because the caller may not be a UI replacement, and may want 'admin' access -- in which case we need to validate roles.